### PR TITLE
Improve support for type inheritance from other mapped types

### LINKED ISSUE #249: MissingReturnAnnotationError
## Describe the Bug

Receiving the following error:

**strawberry.exceptions.missing_return_annotation.MissingReturnAnnotationError**

When trying to use inheritance with the Strawberry models and SQLAlchemy model relationships.  I have a very contrived example below but while the error is a bit confusing i'm not sure what the desired behaviour should be.
Should the mapper be able to handle inheritance?

```python
from sqlalchemy.orm import (
    DeclarativeBase,
    relationship,
    Mapped,
    mapped_column,
)
from sqlalchemy import ForeignKey
 
from strawberry_sqlalchemy_mapper import StrawberrySQLAlchemyMapper
 
strawberry_sqlalchemy_mapper = StrawberrySQLAlchemyMapper()
 
class Base(DeclarativeBase):
    pass
 
class ModelA(Base):
    __tablename__ = "a"
 
    id: Mapped[str] = mapped_column(primary_key=True)
 
    relationshipB_id: Mapped[str] = mapped_column(ForeignKey(f"b.id"))
    relationshipB: Mapped["ModelB"] = relationship()
 
class ModelB(Base):
    __tablename__ = "b"
 
    id: Mapped[str] = mapped_column(primary_key=True)
 
    relationshipB_id: Mapped[str] = mapped_column(ForeignKey(f"b.id"))
    relationshipB: Mapped["ModelB"] = relationship()
 
@strawberry_sqlalchemy_mapper.type(ModelA)
class ApiA:
    pass
 
@strawberry_sqlalchemy_mapper.type(ModelB)
class ApiB(ApiA):
    pass
```

## System Information

 - Operating system: windows
 - Strawberry version (if applicable): 0.268.1 / 0.6.1

### PR DESCRIPTION
**Pull request by Ckk3**
## Description

Ensures consistent behavior when extending mapped types, resolving conflicts between inherited and model-defined fields more predictably.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #249 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

## Summary by Sourcery

Improve support for type inheritance by centralizing generated resolver handling, merging annotations appropriately for different Python versions, and updating documentation and tests to cover inheritance, exclusion, and conflict resolution

Bug Fixes:
- Ensure consistent inheritance behavior for mapped types by resolving conflicts between inherited and model-defined fields and merging annotations predictably across Python versions

Enhancements:
- Introduce a helper to centralize wrapping of generated resolvers from both mapped and original types

Documentation:
- Add a Type Inheritance section to the README with examples on inheritance, exclusion, conflict resolution, and overrides
- Include RELEASE.md detailing patch release notes for inheritance improvements

Tests:
- Add tests covering inheritance scenarios including basic inheritance, exclusion of fields, duplicate field conflicts, and explicit overrides

### CODE REVIEW

File: src/strawberry_sqlalchemy_mapper/mapper.py [hunk 1]
```diff
@@ -798,7 +820,15 @@ def convert(type_: Any) -> Any:
...
-            type_.__annotations__.update(old_annotations)
+
+            # For Python versions <= 3.9, only update annotations that don't already exist
+            # because this versions handle inherance differently
+            if sys.version_info[:2] <= (3, 9):
+                for k, v in old_annotations.items():
+                    if k not in type_.__annotations__:
+                        type_.__annotations__[k] = v
+            else:
+                type_.__annotations__.update(old_annotations)
```
Comment (bellini666) [inline src/strawberry_sqlalchemy_mapper/mapper.py:831]: **question:** what is the exact difference between them? In theory they should be handling that the same

One thing that I know is that in previous versions, `__annotations__` could not exist in the class if that class didn't have any annotations in it, something which I had to workaround on strawberry-django.

Is that it or something else?

**Review by bellini666 — APPROVED** [review]
In general LGTM. Just curious to understand what is the reason for the comment I've made

File: src/strawberry_sqlalchemy_mapper/mapper.py (hunk 1 above)
Comment (Ckk3) [reply src/strawberry_sqlalchemy_mapper/mapper.py:831]: Hi @bellini666, sorry for the delay!
The problem isn’t directly related to `__annotations__` existance, but to inconsistencies with inherited classes and this annotations. 

In Python 3.10 and newer, the `__annotations__`  no longer includes the annotations from `original_type` (base class in this case), which caused some resolvers or type hints to be missing in the final class. So I had to add extra logic to extract those annotations manually.
I didn’t investigate this deeply, but I suspect it only happens because we’re using a decorator, which might interfere with how Python handles class inheritance and type resolution at that point.

On the other hand, in Python 3.8 and 3.9, the inherited annotations are already present in the subclass’s `__annotations__`. So thats why I only add missing keys, since its was raising a error when only using `update()`.

Let me know if you see any better idea to fix this, this one was the most stable solution I found to make it work consistently across versions.
